vassal-app/src/main/java/VASSAL/configure/RefreshPredefinedSetupsDialog.java

```diff
@@ -188,75 +281,295 @@ public void log(String message) {
   }
 
   public boolean isTestMode() {
-    return options.contains("TestMode"); //$NON-NLS-1$
+    return options.contains(GameRefresher.TEST_MODE); //$NON-NLS-1$
   }
 
-
-  private boolean hasAlreadyRun = false;
+  public boolean isFilterMode() {
+    return pdsFilter != null && !pdsFilter.isBlank(); //$NON-NLS-1$
+  }
 
   private void refreshPredefinedSetups() {
-    if (hasAlreadyRun) {
-      return;
-    }
-
-    hasAlreadyRun = true;
-    refreshButton.setEnabled(false);
 
     setOptions();
-    if (isTestMode()) {
-      log(Resources.getString("GameRefresher.refresh_counters_test_mode"));
+
+    // Disable options menu whilst the refresh is assessed (this greys the menu panel out)
+    optionsUserMode = false;
+    for (final Component component : getComponents(this)) component.setEnabled(false);
+
+    // pre-pack regex pattern in case filter string is not found by direct string comparison
+    Pattern filterPattern = null;
+    Pattern filterPattern2 = null;
+
+    if (isFilterMode()) {
+
+      try {
+        // matching, assuming Regex with no escape of the end string modifier, otherwise match all to end
+        filterPattern = Pattern.compile(pdsFilter, CASE_INSENSITIVE);
+        filterPattern2 = Pattern.compile(pdsFilter + ".*", CASE_INSENSITIVE);
+      }
+      catch (java.util.regex.PatternSyntaxException e) {
```

Comment:
Import this. We try to avoid using fully-qualified class names.

Reply:
Done.